### ISSUE TITLE
Running in Windows 

### ISSUE THREAD

**Issue by techsmarty** [andrewning/sortphotos#31]
Works great in Linux. Any clue as to why this isn't running in Windows?

-------
Python 2.7
`Traceback (most recent call last):
File "c:\test\sortphotos.py", line 452, in <module>
    args.use_only_tags, not args.silent)
File "c:\test\sortphotos.py", line 275, in sortPhotos
    with ExifTool() as e:
File "c:\test\sortphotos.py", line 171, in __enter__
    stdin=subprocess.PIPE, stdout=subprocess.PIPE)
File "c:\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
File "c:\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified`

--------
Python 3.4 (Practically the same error)
`Traceback (most recent call last):
  File "c:\test\sortphotos.py", line 452, in <module>
    args.use_only_tags, not args.silent)
  File "c:\test\sortphotos.py", line 275, in sortPhotos
    with ExifTool() as e:
  File "c:\test\sortphotos.py", line 171, in __enter__
    stdin=subprocess.PIPE, stdout=subprocess.PIPE)
  File "c:\Python34\lib\subprocess.py", line 858, in __init__
    restore_signals, start_new_session)
  File "c:\Python34\lib\subprocess.py", line 1111, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The system cannot find the file specified`

---
Seems as though exiftool is installed properly as it executes when run

----
So for Troubleshooting I attempted to change line 21 to from

`exiftool_location =os.path.join(os.path.dirname(os.path.realpath(__file__)), 'Image-ExifTool', 'exiftool')`
to
`exiftool_location = "C:\Windows\System32"`

but came up with this instead: 
PermissionError: [WinError 5] Access is denied

**Comment by PaulRobson** [comment]
Took me a while but I got this working. The main problem was that at first i didn't have perl installed. Once you have installed it, make sure that it's in your path and typing "perl" on the command line starts the interpreter.

I then also had to make some minor code changes, probably because of a difference between how newlines are handled on windows versus linux/osx because the script waited forever after the exit script completed. I changed line 172 to:

    sentinel = "{ready}"

196 to:
    while not output.rstrip(' \t\n\r').endswith(self.sentinel):

and 202 to:
    return output.rstrip(' \t\n\r')[:-len(self.sentinel)]

Maybe worth a pull request?
